.github/actions/setup-pyenv/action.yml

```diff
@@ -0,0 +1,52 @@
+name: "setup-pyenv"
+description: "Setup pyenv"
+runs:
+  using: "composite"
+  steps:
+    ########## Ubuntu ##########
+    - name: Install python build tools
+      if: runner.os == 'Linux'
+      shell: bash
+      # Ref: https://github.com/pyenv/pyenv/wiki#suggested-build-environment
+      run: |
+        sudo apt-get update -y
+        sudo apt-get install -y make build-essential libssl-dev zlib1g-dev \
+        libbz2-dev libreadline-dev libsqlite3-dev wget curl llvm \
+        libncursesw5-dev xz-utils tk-dev libxml2-dev libxmlsec1-dev libffi-dev liblzma-dev
+    - name: Install pyenv
+      if: runner.os == 'Linux'
+      shell: bash
+      run: |
+        git clone --depth 1 https://github.com/pyenv/pyenv.git $HOME/.pyenv
+    - name: Setup environment variables
+      if: runner.os == 'Linux'
+      shell: bash
+      run: |
+        PYENV_ROOT="$HOME/.pyenv"
+        PYENV_BIN="$PYENV_ROOT/bin"
+        echo "$PYENV_BIN" >> $GITHUB_PATH
+        echo "PYENV_ROOT=$PYENV_ROOT" >> $GITHUB_ENV
+    - name: Check pyenv version
+      if: runner.os == 'Linux'
+      shell: bash
+      run: |
+        pyenv --version
```

Comment:
pyenv is successfully installed:

https://github.com/mlflow/mlflow/runs/5537293797?check_suite_focus=true#step:5:298